odl/operator/pspace_ops.py

```diff
@@ -562,6 +606,273 @@ def adjoint(self):
         return ComponentProjection(self.range, self.index)
 
 
+class BroadcastOperator(Operator):
+    """Broadcast argument to set of operators.
+
+    An argument is broadcasted by evaluating several operators in the same
+    point
+
+        ``BroadcastOperator(op1, op2)(x) = [op1(x), op2(x)]``
+
+    It is implemented using a `ProductSpaceOperator`.
+    """
+    def __init__(self, *operators):
+        self._operators = operators
+        self._prod_op = ProductSpaceOperator([[op] for op in operators])
+
+        super().__init__(self.prod_op.domain[0],
+                         self.prod_op.range,
+                         self.prod_op.is_linear)
+
+    @property
+    def prod_op(self):
+        """The prod-op implementation"""
+        return self._prod_op
+
+    @property
+    def operators(self):
+        """A tuple of sub-operators"""
+        return self._operators
+
+    def _call(self, x, out=None):
+        """Apply operators to ``x``.
+
+        Parameters
+        ----------
+        x : domain element
+            Input vector to be evaluated by operators
+        out : range element, optional
+            output vector to write result to
+
+        Returns
+        -------
+        out : range element
+            Values of operators evaluated in point
+
+        Examples
+        --------
+        >>> import odl
+        >>> I = odl.IdentityOperator(odl.Rn(3))
+        >>> op = BroadcastOperator(I, 2 * I)
+        >>> x = [1, 2, 3]
+        >>> op(x)
+        ProductSpace(Rn(3), 2).element([
+            [1.0, 2.0, 3.0],
+            [2.0, 4.0, 6.0]
+        ])
+        """
+        wrapped_x = self.prod_op.domain.element([x], cast=False)
+        return self.prod_op(wrapped_x, out=out)
+
+    def derivative(self, x):
+        """Derivative of the broadcast operator.
+
+        Parameters
+        ----------
+        x : domain element
+            The point to take the derivative in
+
+        Returns
+        -------
+        adjoint : linear `BroadcastOperator`
+            The derivative
+
+        Examples
+        --------
+        >>> import odl
+        >>> I = odl.IdentityOperator(odl.Rn(3))
+        >>> x = [1, 2, 3]
+
+        Example with linear operator (derivative is itself)
+
+        >>> op = BroadcastOperator(I, 2 * I)
+        >>> op(x)
+        ProductSpace(Rn(3), 2).element([
+            [1.0, 2.0, 3.0],
+            [2.0, 4.0, 6.0]
+        ])
+        >>> op.derivative(x)(x)
+        ProductSpace(Rn(3), 2).element([
+            [1.0, 2.0, 3.0],
+            [2.0, 4.0, 6.0]
+        ])
```

Comment:
Can you find an example where the derivative is different from the original operator? Maybe use `ResidualOp`, (in one component), should be simple enough.